File: Data Structures & Algorithms/meeting-schedule-ii/submission-4.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        if len(intervals)==0:
            return 0
        intervals.sort(key = lambda x:x.start)
        i,n=1,len(intervals)
        L=[intervals[0].end]
        while i<n:
            if intervals[i].start < L[0]:
                L.append(intervals[i].end)
            else:
                L[0]=max(intervals[i].end,L[0])
            L.sort()
            i+=1
        return len(L)
        

File: Data Structures & Algorithms/merge-intervals/submission-4.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
         return a[0] < b[0];
        });
        int i=0,n=intervals.size();
        while(i<n-1){
            if(intervals[i][1]>=intervals[i+1][0]){
                intervals[i][1]=max(intervals[i][1],intervals[i+1][1]);
                intervals.erase(intervals.begin() + i+1);
                n=intervals.size();
            }
            else i++;
        }
        return intervals;
    }
};
